### PR TITLE
[Android] pause audio playback when headphone is disconnected

### PR DESCRIPTION
**Pull request by oldk1331**
## Description

On Android phones, when headphone gets disconnected, kodi will pause the playback.

## Motivation and Context

It might be awkward if your android phone is still playing music (loudly) when you disconnected your headphone (maybe unaware of bluetooth goes out of range). This is a feature in many other android players.

## How Has This Been Tested?

I build kodi on top of master branch with my change, install it on my android phone, play a music file with headphone plugged in, then unplug it and kodi pauses music playback.  The same goes for video playback.

The change is pretty simple and should not affect any other parts of kodi.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fritsch** [comment]
Thanks much, pinged our Android Maintainer for review!

File: xbmc/platform/android/activity/XBMCApp.cpp
Comment (Rechi) [inline xbmc/platform/android/activity/XBMCApp.cpp]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/platform/android/activity/XBMCApp.cpp
Comment (oldk1331) [reply xbmc/platform/android/activity/XBMCApp.cpp]: Updated by using clang-format to format this chunk of code.

**Comment by oldk1331** [comment]
Hello maintainers, can you add labels please? I'd like to see this simple change backported to Leia if possible.

**Comment by oldk1331** [comment]
Fixed a bug that when music is paused while headphone is connected, unplug it and music will play.  Fix it by checking `PLAYBACK_STATE_PLAYING` as well.

**Comment by phunkyfish** [comment]
@peak3d would you have time to review?

**Review by peak3d — APPROVED** [review]
Looks good to me! Thx @oldk1331

**Comment by peak3d** [comment]
Jenkins build this please

**Comment by oldk1331** [comment]
Hi, how can I help to move this forward, make a rebase or something?

**Comment by phunkyfish** [comment]
I don’t think anything else is needed will merge shortly.

**Comment by phunkyfish** [comment]
You can prepare the backport if you like.

**Comment by DaveTBlake** [comment]
As an improvement not a fix this should not have been backported. It is also causing a regression, hence the backport will be reverted (one jenkins gives green).  @matthuisman can you look at whatever needs to happen on master for this feature to work correctly

**Comment by matthuisman** [comment]
@DaveTBlake
Did you want this reverted on Matrix as well?

Can you "update" a PR once it's merged?
Or would the author need to create a new PR?
In that case - best to revert and wait for new PR?

**Comment by DaveTBlake** [comment]
No, PRs do not update and not so fast on reverting master.  IMO there is no need for am immediate revert on master, we can give the author chance to respond. Also it could be a single PR that undoes this and implements the feature differently, that could be from @oldk1331 or someone else.  I would prefer to give the author chance to work on this first or say that they can't.

**Comment by oldk1331** [comment]
@matthuisman Can you open a new issue so that we can move discussion there?

**Comment by matthuisman** [comment]
@oldk1331 
I have a draft PR of what seems to work for me.

Should be fine to discuss there:
https://github.com/xbmc/xbmc/pull/17913

**Pull request merged**
